Thanks @wentasah!  I would be happy to merge this in.  The CI picked up an error when running pyflakes:

https://github.com/ros-infrastructure/superflore/actions/runs/18121962057/job/51703339669?pr=326#step:5:96
```
b"/home/runner/work/superflore/superflore/tests/../superflore/generators/nix/run.py:26:1: F401 'superflore.repo_instance.RepoInstance' imported but unused\n1\n"
```

Would you be open to adding a commit to the PR to resolve this issue?

--- a/superflore/generators/nix/run.py
+++ b/superflore/generators/nix/run.py
@@ -58,7 +58,11 @@ def main():
         if not args.output_repository_path:
             parser.error('Invalid args! no repository specified')
         try:
-            prev_overlay = NixRosOverlay(args.output_repository_path, False)
+            prev_overlay = NixRosOverlay(
+                args.output_repository_path,
+                False,
+                from_branch=args.upstream_branch,
+            )
             msg, title = load_pr()
             prev_overlay.pull_request(msg, title=title)
             clean_up()

--- a/superflore/generators/nix/run.py
+++ b/superflore/generators/nix/run.py
@@ -23,7 +23,6 @@ from superflore.generators.nix.gen_packages import regenerate_pkg, \
     regenerate_pkg_set
 from superflore.generators.nix.nix_ros_overlay import NixRosOverlay
 from superflore.parser import get_parser
-from superflore.repo_instance import RepoInstance
 from superflore.TempfileManager import TempfileManager
 from superflore.utils import clean_up, get_distros_by_status
 from superflore.utils import err
@@ -59,7 +58,7 @@ def main():
         if not args.output_repository_path:
             parser.error('Invalid args! no repository specified')
         try:
-            prev_overlay = RepoInstance(args.output_repository_path, False)
+            prev_overlay = NixRosOverlay(args.output_repository_path, False)
             msg, title = load_pr()
             prev_overlay.pull_request(msg, title=title)
             clean_up()

